ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Hest/HestParser.c

```diff
@@ -0,0 +1,958 @@
+             Ptr,
+             Size,
+             PARSER_PARAMS (HestErrorNotificationParser)
+             );
+  if (Offset != Size) {
+    IncrementErrorCount ();
+    Print (L"ERROR: Failed to parse Hardware Error Notification Structure!\n");
+  }
+}
+
+/**
+  Dumps bus field in the PCI related Error Source Structure.
+                                      from HestTable.
+  @param [in] Format  Optional format string for tracing the data.
+  @param [in] Ptr     Pointer to the start of the buffer.
+**/
+STATIC
+VOID
+EFIAPI
+DumpPciBus (
+  IN CONST CHAR16  *Format OPTIONAL,
+  IN UINT8         *Ptr
+  )
+{
+  if (Format != NULL) {
+    Print (Format, *(UINT32 *)Ptr);
+    return;
+  }
+
+  Print (L"0x%x\n", *Ptr);
+  ParseAcpiBitFields (
+    TRUE,
+    2,
+    NULL,
+    Ptr,
+    1,
+    PARSER_PARAMS (HestErrorSourcePciCommonBusParser)
+    );
+}
+
+/**
+  Dumps the IA32 Arch Machine Check Error Bank structure fields.
+
+  @param [in]       HestTable         Start pointer to Hest table.
+  @param [in]       AcpiTableLength   Length of HestTable.
+  @param [in,out]   Offset            Offset to machine check bank structure
+                                      from HestTable.
+
+  @retval EFI_SUCCESS                   Success
+  @retval EFI_INVALID_PARAMETER         Invalid Hest Table
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+DumpIA32ArchMachineCheckErrorBankStructure (
+  IN UINT8  *HestTable,
+  UINT32    AcpiTableLength,
+  UINT32    *Offset
+  )
+{
+  UINT8   Idx;
+  UINT8   *IA32BankStructPtr;
+  UINT32  TotalBankStructSize;
+
+  TotalBankStructSize = *mHestIA32HardwareBankCount *
+                        sizeof (EFI_ACPI_6_5_IA32_ARCHITECTURE_MACHINE_CHECK_ERROR_BANK_STRUCTURE);
+
+  if ((*Offset + TotalBankStructSize) > AcpiTableLength) {
+    IncrementErrorCount ();
+    Print (
+      L"ERROR: Not enough data for "
+      "IA-32 Architecture Machine Check Exception Error source.\n"
+      );
+    return EFI_INVALID_PARAMETER;
+  }
+
+  for (Idx = 0; Idx < *mHestIA32HardwareBankCount; Idx++) {
+    IA32BankStructPtr = HestTable + *Offset;
+    ParseAcpi (
+      TRUE,
+      4,
+      "IA-32 Architecture Machine Check Bank Structure",
+      IA32BankStructPtr,
+      sizeof (EFI_ACPI_6_5_IA32_ARCHITECTURE_MACHINE_CHECK_ERROR_BANK_STRUCTURE),
+      PARSER_PARAMS (HestErrorIA32ArchMachineCheckBankStructureParser)
+      );
+    *Offset +=
+      sizeof (EFI_ACPI_6_5_IA32_ARCHITECTURE_MACHINE_CHECK_ERROR_BANK_STRUCTURE);
+  }
+
+  *mHestIA32HardwareBankCount = 0;
+
+  return EFI_SUCCESS;
+}
+
+/**
+  Helper macro to populate the header fields of error source descriptor in the
+  ACPI_PARSER array.
+**/
+#define PARSE_HEST_ERROR_SOURCE_COMMON_HEADER(FlagsValidateFunc, CheckRecordCount)   \
+  { L"Type", 2, 0, L"%d", NULL, NULL, NULL, NULL },                           \
+  { L"Source Id", 2, 2, L"%d", NULL, NULL, NULL, NULL },                      \
+  { L"Reserved", 2, 4, NULL, NULL, NULL, NULL, NULL },                     \

```

Comment:
Uncrustify doesn't seem to complain about this,
but would it be possible to align the ending '\' of the macros (similar comments for macros below)

Reply:
Thanks! I'll fix it :)